crates/kumod/src/queue.rs

```diff
@@ -142,6 +142,29 @@ impl ScheduledMetrics {
         self.by_tenant.as_ref().map(|m| m.sub(amount));
         self.by_tenant_campaign.as_ref().map(|m| m.sub(amount));
     }
+
+    pub fn remove_metrics_for_queue(name: &str) {
+        let components = QueueNameComponents::parse(name);
+
+        DELAY_GAUGE.remove_label_values(&[name]).ok();
+
+        match &components.campaign {
+            Some(campaign) => Some(TENANT_CAMPAIGN_GAUGE.remove_label_values(&[
+                components.tenant.as_ref().map(|s| s.as_ref()).unwrap_or(""),
+                campaign,
+            ])),
+            None => None,
+        };
```

Comment:
I deliberately omitted this because it would remove too much: this particular counter aggregates across multiple scheduled queues, so it shouldn't be removed by any one scheduled queue being aged out.

I think that we don't strictly need to remove metrics for this one, because there should be a reasonably well bounded combination of tenants and campaigns; the cardinality is much smaller than the potential maximum number of email domains, for example.

If we were to age it out, I think that there would need to be some kind of periodic background task that checked for all combinations of tenant+campaign to decide when it would be appropriate to remove any given instance of this counter.

```suggestion
```

Reply:
ah yeah, my bad... I forgot that metrics of routing domains were also included.